server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/MessageAppender.java

```diff
@@ -0,0 +1,113 @@
+
+    private final AttachmentManager attachmentManager;
+    private final MIMEMessageConverter mimeMessageConverter;
+
+    @Inject
+    public MessageAppender(AttachmentManager attachmentManager, MIMEMessageConverter mimeMessageConverter) {
+        this.attachmentManager = attachmentManager;
+        this.mimeMessageConverter = mimeMessageConverter;
+    }
+
+    public MessageFactory.MetaDataWithContent createMessageInMailbox(ValueWithId.MessageWithId.CreationMessageEntry createdEntry,
+                                                                     MessageManager mailbox,
+                                                                     MailboxSession session) throws MailboxException {
+        ImmutableList<MessageAttachment> messageAttachments = getMessageAttachments(session, createdEntry.getValue().getAttachments());
+        byte[] messageContent = mimeMessageConverter.convert(createdEntry, messageAttachments);
+        SharedByteArrayInputStream content = new SharedByteArrayInputStream(messageContent);
+        Date internalDate = Date.from(createdEntry.getValue().getDate().toInstant());
+
+        Keywords keywords = createdEntry.getValue()
+            .getKeywords()
+            .orElse(Keywords.DEFAULT_VALUE);
+        boolean notRecent = false;
+
+        ComposedMessageId message = mailbox.appendMessage(content, internalDate, session, notRecent, keywords.asFlags());
+
+        return MessageFactory.MetaDataWithContent.builder()
+            .uid(message.getUid())
+            .keywords(keywords)
+            .internalDate(internalDate.toInstant())
+            .sharedContent(content)
+            .size(messageContent.length)
+            .attachments(messageAttachments)
+            .mailboxId(mailbox.getId())
+            .messageId(message.getMessageId())
+            .build();
+    }
+
+    private ImmutableList<MessageAttachment> getMessageAttachments(MailboxSession session, ImmutableList<Attachment> attachments) throws MailboxException {
+        ThrowingFunction<Attachment, Optional<MessageAttachment>> toMessageAttachment = att -> messageAttachment(session, att);
+        return attachments.stream()
+            .map(Throwing.function(toMessageAttachment).sneakyThrow())
+            .flatMap(OptionalUtils::toStream)
+            .collect(Guavate.toImmutableList());
+    }
+
+    private Optional<MessageAttachment> messageAttachment(MailboxSession session, Attachment attachment) throws MailboxException {
+        try {
+            return Optional.of(MessageAttachment.builder()
+                .attachment(attachmentManager.getAttachment(AttachmentId.from(attachment.getBlobId().getRawValue()), session))
+                .name(attachment.getName().orElse(null))
+                .cid(attachment.getCid().map(Cid::from).orElse(null))
+                .isInline(attachment.isIsInline())
+                .build());
+        } catch (AttachmentNotFoundException e) {
+            // should not happen (checked before)
```

Comment:
is this comment still true?

Reply:
Probably not, but it can be safely removed I assume.